x-pack/elastic-agent/pkg/composable/providers/agent/agent.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package agent
+
+import (
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/application/info"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/composable"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/config"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release"
+)
+
+func init() {
+	composable.Providers.AddContextProvider("agent", ContextProviderBuilder)
```

Comment:
I wonder if we should call it `elastic-agent` to make it more descriptive? It's not as nice in the config but I expect it to never show up, so probably does not matter.

Reply:
If we call it that, then in the variable you would need to use `{{elastic-agent.id}}`, don't know if I like that better than `{{agent.id}}`.